Operator Panic

**Describe the bug**
When the external certificate is invalid, the operator will be in a crash loop.

```
2019-09-19T07:54:18.172Z	DEBUG	controllers.KafkaCluster	resource is in sync	{"kafkacluster": "kafka/kafka", "Request.Name": "kafka", "component": "kafka", "kind": "*v1.Service", "name": "kafka-headless"}
E0919 07:54:18.172914       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:76
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:65
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:679
/usr/local/go/src/runtime/panic.go:199
/usr/local/go/src/runtime/signal_unix.go:394
/workspace/pkg/certutil/main.go:72
/workspace/pkg/resources/kafka/kafka.go:233
/workspace/controllers/kafkacluster_controller.go:109
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:216
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1357
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1177be4]
```


**Steps to reproduce the issue:**
Use invalid cert as external cert.

**Expected behavior**
The operator shouldn't restart continuously.

--- a/.licensei.toml
+++ b/.licensei.toml
@@ -30,6 +30,7 @@ ignored = [
   "github.com/prometheus/procfs",
   "github.com/google/uuid",
   "github.com/pborman/uuid",
+  "github.com/prometheus/client_golang",
 
   "github.com/davecgh/go-spew", # ISC license
 ]

--- a/pkg/certutil/main.go
+++ b/pkg/certutil/main.go
@@ -19,6 +19,7 @@ import (
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
+	"errors"
 	"math/rand"
 	"strings"
 	"text/template"
@@ -49,6 +50,9 @@ ssl.key.password={{ .Password }}
 
 func DecodeKey(raw []byte) (parsedKey []byte, err error) {
 	block, _ := pem.Decode(raw)
+	if block == nil {
+		return nil, errors.New("failed to decode PEM block")
+	}
 	var keytype certv1.KeyEncoding
 	var key interface{}
 	if key, err = x509.ParsePKCS1PrivateKey(block.Bytes); err != nil {
@@ -70,6 +74,9 @@ func DecodeKey(raw []byte) (parsedKey []byte, err error) {
 
 func DecodeCertificate(raw []byte) (cert *x509.Certificate, err error) {
 	block, _ := pem.Decode(raw)
+	if block == nil {
+		return nil, errors.New("failed to decode PEM block")
+	}
 	cert, err = x509.ParseCertificate(block.Bytes)
 	return
 }

